README.md

```diff
@@ -100,17 +102,89 @@ Below are the steps for running and performing data analysis on your local machi
    should be analysed qualitatively by the data scientist for each category & sector (there are over 150 different 
    combination) before finalising the best value.
 
+-------------------------------------------------------
+### Data pipeline
+
+Currently there are 3 runnable data pipeline jobs in this repo.
+Each job can be runned as a separate task, or orchestrated in Azure DataFactory to run as a complete workflow.
+
+#### Before running pipeline jobs
+If you are running the jobs on a local machine, ensure that you fill in blob account_name and blob access_key in file `config.ini`. Also, run the command `az login` beforehead to authenticate yourself with azure cloud.
+
+If you are running the jobs on Azure cloud environment, ensure infrastructures in repo `Crown-Commercial-Service/ccs-spend-forecasting-infra` and `Crown-Commercial-Service/ccs-spend-forecasting-adf` are properly setup already.
+
+#### 1. Fill missing months to spend data
+Related files: [`pipeline/jobs/fill_missing_months.py`, `pipeline/main.py`]
+
+This job insert dummy records of amount £0 spending into spend data for in-between months that has no spending.
+
+For every Category/Market sector combinations, it will find the earliest and most recent dates of spend record, and within the period if there is any month that doesn't have any record, it will insert a dummy record with amount zero.
+
+This jobs requires the `TransformedSpendData` data file in blob storage, which is generated by a pipeline job in Azure Data Factory.
+The output is stored on blob storage in folder  `SpendDataFilledMissingMonth`.
+
+##### To run this job on a local machine:
+```shell
+poetry run python -m pipeline.main
```

Comment:
Worth adding to the `Makefile` and amending to `make pipeline_local` or similar?